sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -204,7 +204,7 @@ object JDBCRDD extends Logging {
  */
 private[jdbc] class JDBCRDD(
     sc: SparkContext,
-    getConnection: () => Connection,
+    getConnection: JDBCPartition => Connection,
```

Comment:
What partition does spark pass to this function if spark wants to create a connection at the driver side?

Reply:
Users could construct JDBCPartition by self way. Such as: set the partition index with shard node list.

Reply:
This did not answer the question.

Reply:
Yeah. In general, Spark create a connection at Driver side no need Partition or not know which Partition is.